force-app/main/default/classes/Gifts.cls

```diff
@@ -59,6 +59,15 @@ public with sharing class Gifts {
         setElevateBatchIds(dataImports);
     }
 
+    public Gift findBy(GiftId giftId) {
+        for (DataImport__c dataImport : dataImports) {
```

Comment:
Would it be worth storing dataImports internally as a Map<Id,DataImport__c> so that we don't have to do this iteration?